### ISSUE TITLE
Nodata masking bug in GLOBIO

### ISSUE THREAD

**Issue by dcdenu4** [natcap/invest#122]
There is a nodata masking bug that was brought to light from debugging a users data from this forum post: [natcap-forums:globio-unexpected-output](https://community.naturalcapitalproject.org/t/globio-unexpected-output/978/4)

The bug occurs in the following function:
```
def _collapse_infrastructure_op(*infrastructure_array_list):
        """For each pixel, create mask 1 if all valid, else set to nodata."""
        nodata_mask = (
            numpy.isclose(
                infrastructure_array_list[0], infrastructure_nodata_list[0]))
        infrastructure_result = infrastructure_array_list[0] > 0
        for index in range(1, len(infrastructure_array_list)):
            current_nodata = numpy.isclose(
                infrastructure_array_list[index],
                infrastructure_nodata_list[index])

            infrastructure_result = (
                infrastructure_result |
                ((infrastructure_array_list[index] > 0) & ~current_nodata))

            nodata_mask = (
                nodata_mask & current_nodata)

        infrastructure_result[nodata_mask] = infrastructure_nodata
        return infrastructure_result
```

When `*infrastructure_array_list` is of length 1, `infrastructure_result` is set to a boolean type. Then before returning the result is masked and attempted to be updated with an integer type of `-1`. This causes improper masking.

**Comment by dcdenu4** [comment]
Helpful information from Rich (referring to this gist https://gist.github.com/dcdenu4/b7bcb9c317869ced37b46426bfb57da6):
>OH I thought the issue was that line 13 defines result as a Boolean array then in line 17 gets a non-boolean assignment to it, specifically -1. I don't read that as an error with a Boolean array indexing a Boolean array, but rather a type error. Since it's Python and numpy, the -1 will get cast to True (since `bool(-1)` is "True") so line 17 is the same as `result[nodata_mask] = True`. I suspect that is an unexpected result.

**Comment by dcdenu4** [comment]
For myself to remember: I have a branch in my fork `bugfix/globio-numpy-bool-bug`

**Comment by dcdenu4** [comment]
resolved in PR #186
